File: tests/fixtures/b2t_typescript_closed_state/before/src/jobs.ts
export type JobState = "queued" | "running" | "done";

export interface Job {
  state: JobState;
}

export function queue(job: Job): void {
  job.state = "queued";
}

export function isRunning(job: Job): boolean {
  return job.state === "running";
}

export function isDone(job: Job): boolean {
  return "done" === job.state;
}

export function isQueuedAlias(job: Job): boolean {
  const currentState = job.state;
  return currentState === "queued";
}

export function ensureQueued(job: Job): void {
  job.state ??= "queued";
}

export function queueBoth(primary: Job, backup: Job): void {
  primary.state = backup.state = "queued";
}

File: tests/fixtures/b2p_state_reference/after/app/services/jobs.py
from app.models.jobs import Job, JobStatus


def start(job: Job) -> bool:
    job.status = JobStatus.QUEUED
    return job.status == JobStatus.QUEUED


def can_finish(job: Job) -> bool:
    return job.status in (JobStatus.QUEUED, JobStatus.RUNNING)


def is_done(job: Job) -> bool:
    return JobStatus.DONE == job.status
